src/Symfony/Component/DependencyInjection/Compiler/AutowireRequiredPropertiesPass.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Looks for definitions with autowiring enabled and registers their corresponding "@required" properties.
+ *
+ * @author Sebastien Morel (Plopix) <morel.seb@gmail.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class AutowireRequiredPropertiesPass extends AbstractRecursivePass
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function processValue($value, bool $isRoot = false)
+    {
+        if (\PHP_VERSION_ID < 70400) {
+            return $value;
+        }
+        $value = parent::processValue($value, $isRoot);
+
+        if (!$value instanceof Definition || !$value->isAutowired() || $value->isAbstract() || !$value->getClass()) {
+            return $value;
+        }
+        if (!$reflectionClass = $this->container->getReflectionClass($value->getClass(), false)) {
+            return $value;
+        }
+
+        foreach ($reflectionClass->getProperties() as $reflectionProperty) {
+            if (!$reflectionProperty->isPublic()) {
```

Comment:
let's remove this check: AutowireRequiredMethodsPass doesn't have it - it will trigger an error at runtime, which would be legit

Reply:
ok done